openedx/core/lib/block_structure/block_structure.py

```diff
@@ -564,7 +564,7 @@ def remove_transformer_block_field(self, usage_key, transformer, key):
         """
         try:
             transformer_block_data = self.get_transformer_block_data(usage_key, transformer)
-            delattr(transformer_block_data, key)
+            del transformer_block_data.fields[key]
```

Comment:
I think you may be calling it differently from what it's expecting.
Unfortunately, the interface is not consistent.  In some cases we require clients to use the `getattr` calls (as class member attributes), and in other cases `.get()` calls (as dict access).